### PR TITLE
define the ietf-schedule module in a seperate draft, move it out from this draft

### PR DESCRIPTION
**Pull request by QiufangMa**
see https://github.com/QiufangMa/ietf-schedule-yang for the new draft #17

### CODE REVIEW

**Review by boucadair — APPROVED** [review]

**Comment by boucadair** [comment]
> see https://github.com/QiufangMa/ietf-schedule-yang for the new draft #17

I guess you should update the new repo setup to allow read-write github actions. 

If you struggle with that, a simple option would be to add the md file to the existing repo.

**Comment by QiufangMa** [comment]
> > see https://github.com/QiufangMa/ietf-schedule-yang for the new draft #17
> 
> I guess you should update the new repo setup to allow read-write github actions.
> 
> If you struggle with that, a simple option would be to add the md file to the existing repo.

Thanks Med, I think it should be fixed now, let me know if not.

**Comment by QiufangMa** [comment]
I think the schedule work should be dicussed in NETMOD, but don't have strong feelings about that, would like to learn your thoughts as well.

**Comment by boucadair** [comment]
> I think the schedule work should be dicussed in NETMOD, but don't have strong feelings about that, would like to learn your thoughts as well.

The UCL will have a normative dependency on the schedule module. It should be normal to maintain this in opsawg. I would avoid netmod.

File: draft-ma-netmod-schedule-yang.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+--- middle
+
+# Introduction {#intro}
+
+   This document defines a common schedule YANG module which
+   is designed to be applicable for scheduling information such as event, policy,
+   services or resources based on date and time. This model is designed with the intent to be reusable in other
+   scheduling contexts. In particular, the groupings can be reused by other modules.
+
+   The YANG data models in this document conform to the Network
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:70]: I would remove this mention as the module is about groupings and types.

File: draft-ma-netmod-schedule-yang.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+      { "weekday": "tuesday" },
+      { "weekday": "sunday" }
+    ],
+    "wkst": "monday"
+  }
+}
+~~~
+
+
+#  YANG Modules
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:252]:
```suggestion

```

File: draft-ma-netmod-schedule-yang.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+##  The "ietf-schedule" YANG Module {#sec-schedule}
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:254]:
```suggestion
#  The "ietf-schedule" YANG Module {#sec-schedule}
```

File: draft-ma-netmod-schedule-yang.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+   modules.  The module by itself does not expose any data nodes that
+   are writable, data nodes that contain read-only state, or RPCs.  As
+   such, there are no additional security issues related to the "ietf-
+   schedule" module that need to be considered.
+
+#  IANA Considerations
+
+##  The "IETF XML" Registry
+
+   This document registers the following URIs in the "IETF XML Registry" {{!RFC3688}}.
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:291]:
```suggestion
   This document registers the following URI in the "IETF XML Registry" {{!RFC3688}}.
```

File: draft-ma-netmod-schedule-yang.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+
+~~~~
+        URI: urn:ietf:params:xml:ns:yang:ietf-schedule
+        Registrant Contact: The IESG.
+        XML: N/A, the requested URI is an XML namespace.
+~~~~
+
+##  The "YANG Module Names" Registry
+
+   This document registers the following YANG modules in the "YANG Module Names"
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:301]:
```suggestion
   This document registers the following YANG module in the "YANG Module Names"
```

File: draft-ma-netmod-schedule-yang.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+--- back
+
+# Changes between Revisions
+
+
+# Acknowledgments
+{:numbered="false"}
+
+   This work is derived from the {{?I-D.ietf-opsawg-ucl-acl}}. There is a desire
+   from the WG to see this model be seperately defined for wide use in scheduling context.
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:321]:
```suggestion
   from the OPSAWG to see this model be separately defined for wide use in scheduling context.
```

File: draft-ma-netmod-schedule-yang.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+
+   The authors would like to thank Adrian Farrel, Wei Pan, Tianran Zhou and Joe Clarke
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:323]:
```suggestion
   The authors would like to thank Adrian Farrel, Wei Pan, Tianran Zhou and Joe Clarke
```
```suggestion
   Thanks to Adrian Farrel, Wei Pan, Tianran Zhou, and Joe Clarke
```

File: draft-ma-netmod-schedule-yang.md [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+   for their valuable comments and great input to this work.
```
Comment (boucadair) [inline draft-ma-netmod-schedule-yang.md:324]:
```suggestion
   for their valuable comments and inputs to this work.
```

File: draft-ietf-opsawg-ucl-acl.md [hunk 9]
```diff
@@ -587,29 +419,17 @@ informative:
...
-{::include ./yang/ietf-schedule.yang}
-<CODE ENDS>
-~~~~
-
-
 ##  The "ietf-ucl-acl" YANG Module {#sec-UCL}
 
    This module imports types and groupings defined in the "ietf-schedule" YANG
-   module in {{sec-schedule}}. It also augments the IETF ACL YANG module defined in {{!RFC8519}}.
+   module in <!--to add a reference-->. It also augments the IETF ACL YANG module defined in {{!RFC8519}}.
```
Comment (boucadair) [inline draft-ietf-opsawg-ucl-acl.md:432]: Will need to add a normative reference as well.

**Review by boucadair — CHANGES_REQUESTED** [review]

File: draft-ma-opsawg-schedule-yang.md [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+    email: d.king@lancaster.ac.uk
+
+normative:
+
+informative:
+
+
+--- abstract
+
+   this document defines a common schedule YANG module which is
```
Comment (boucadair) [inline draft-ma-opsawg-schedule-yang.md:57]:
```suggestion
   This document defines a common schedule YANG module which is
```

File: draft-ma-opsawg-schedule-yang.md [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+   designed to be applicable for scheduling information such as event, policy,
+   services or resources based on date and time.
```
Comment (boucadair) [inline draft-ma-opsawg-schedule-yang.md:59]:
```suggestion
   services, or resources based on date and time.
```

File: draft-ma-opsawg-schedule-yang.md [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+
+--- middle
+
+# Introduction {#intro}
+
+   This document defines a common schedule YANG module which
+   is designed to be applicable for scheduling information such as event, policy,
+   services or resources based on date and time.
+
+   This document uses the YANG terminology defined in Section 3 of {{!RFC7950}}.
```
Comment (boucadair) [inline draft-ma-opsawg-schedule-yang.md:69]:
```suggestion
   
```

File: draft-ma-opsawg-schedule-yang.md [hunk 13]
```diff
@@ -0,0 +1,320 @@
...
+   * YYYY --> the assigned RFC number for {{I-D.ietf-netmod-rfc6991-bis}}
+   * 2023-01-19 --> the actual date of the publication of this document
+
+# Conventions and Definitions
+
+{::boilerplate bcp14-tagged}
+
+   The meanings of the symbols in tree diagrams are defined in
+   {{?RFC8340}}.
+
```
Comment (boucadair) [inline draft-ma-opsawg-schedule-yang.md:92]:
```suggestion
Also, this document uses the YANG terminology defined in {{Section 3 of !RFC7950}}.

```

File: draft-ma-opsawg-schedule-yang.md [hunk 14]
```diff
@@ -0,0 +1,320 @@
...
+#  Modules Overview
+
+##  The Schedule YANG Module
+
+   This module defines two groupings "period" and "recurrence", which conform to
+   the definition of the "period of time" and "recurrence rule" formats defined in
+   {{?RFC5545}}. The authors might consider to define more formats in future revisions.
```
Comment (boucadair) [inline draft-ma-opsawg-schedule-yang.md:99]:
```suggestion
   {{?RFC5545}}. Other formats may be considered in future revisions.
```

**Review by boucadair — CHANGES_REQUESTED** [review]

**Pull request merged**